### DISCUSSION TITLE
Particles : Grass not working ?

### DISCUSSION THREAD

**Discussion by ABelliqueux** [tahoma2d/tahoma2d#1030] [Q&A] [answered]
Hi, 

Not sure if it's the right place to ask, but I'm reluctant to open an issue as I'mnot sure it's a bug yet...

In **tahoma2d, latest release on Linux**, I can use all particles effects fine, except for the grass, which seems to never end calculating ( the frame is never greenlit ). What's more, its not red either, it seems like the image keeps being recalculated and the color is a mix of red and green.

![t2d-grass](https://user-images.githubusercontent.com/1405255/185761215-b33cb168-0edb-4c5c-9f03-abb72cf3fd58.png)
 
Can someone confirm this is not working ? Or am I doing something wrong ?

FWIW, with the exact same FX schematic setup, every other particles types work.

**Comment by andeon** [comment, marked as answer]
Double click on the Grass node in the first "Source" tab, put the number 0 in the "Control Image" field to not use a control image.

To know more about: https://tahoma2d.readthedocs.io/en/latest/using_the_particles_effect.html#defining-control-images

I think the default value to create could be 0, but we'd need to check if the effect is used more with a control image.

**Reply by ABelliqueux** [reply]
Great, **it's working with 0 as Control Image**. Although I didn't mention it earlier, it was working with a control image in my case, but I still think the default should be 0 too, as using a source image is a bit more involved and having an effect not working with the defaults settings is a bit confusing...

**Comment by ABelliqueux** [comment]
FWIW, I added a note to the doc and made a PR to change the default preset : 

https://github.com/tahoma2d/tahoma2d_docs/pull/7

https://github.com/tahoma2d/tahoma2d/pull/1034

Feel free to keep only the relevant PR ( or none :) )

**Comment by manongjohn** [comment]
Question...when not working in Implicit Hold mode, does this work properly with the current defaults.

**Comment by ABelliqueux** [comment]
You mean like this ?

![2022-08-25-184052_1920x1043_scrot](https://user-images.githubusercontent.com/1405255/186722273-f1bda71f-6179-4963-bf66-b654b6924270.png)

Rendering seems to hang also ( contrary to what the screenshot suggests, the rendering status of theframe is not green, it keeps on flashing... (And the rendered image on the screenshot was generated with a value of 0 in 'Control Image' )

Let me know if you want a video.

**Comment by manongjohn** [comment]
Yes. Thanks. Curious if it was fine in 1.2 or if it was always broken like this.

**Reply by ABelliqueux** [reply]
Ok, I'll record one now. Fwiw, it does the same thing in opentoonz.

**Reply by manongjohn** [reply]
Oh. So its an existing issue. That's kind of good to know. Thanks for following up. I'll take a look at this and the PR as soon as I am able to.

**Reply by ABelliqueux** [reply]
Maybe I should open an issue on the opentoonz repo then ? And make a PR there ? I really only changed a value in the preset file for that specific effect.

**Reply by manongjohn** [reply]
That's up to you. I regularly pull OT changes into T2D so doing it there will eventually update both systems.

Most T2D changes I make, do not get pulled into OT unless someone wants to port it over.

**Comment by ABelliqueux** [comment]
Here it is : 

https://user-images.githubusercontent.com/1405255/186726705-8cb64788-0cb8-4a2a-9d78-cf1c7b2010aa.mp4

Hopefully it shows the behaviour I'm describing...
